web3indexer/block_fetcher.py

```diff
@@ -15,35 +17,36 @@ class BlockFetcher:
 
     MAX_RETRIES = 5
 
-    def fetch_with_retry(self, dispatcher, w3, task):
+    def __init__(self, w3: Web3):
+        self.w3 = w3
+        self.producer = KafkaProducer(
```

Comment:
NIT: It will make it much easier to test (not that we rigidly enforce testing :disappointed:) if you instantiate this outside of the class and pass it in as an argument to the `__init__` method. That way during unit testing we can just pass in a mock object rather than having to patch.